README.md

```diff
@@ -14,18 +14,38 @@ Implement the tool `st` that does exact pattern matching using a suffix tree. Te
 # Report
 
 ## Specify if you have used a linear time or quadratic time algorithm.
-
+In the implementation we used the quadratic time algorithm. 
 ## Insights you may have had while implementing and comparing the algorithms.
 
 ## Problems encountered if any.
+Not explicit return statement caused very weird problems with None being returned in certain cases.Apart from that using many different variables for indexing into the suffix, edge and the whole string caused some confusion and incorrect results. 
```

Comment:
Type annotations would tell you about returning `None` if you don't expect to. Keeping track of indices is just a pain that can be hard to avoid.